tests/runner/testresource.go

```diff
@@ -90,29 +91,65 @@ func (r *TestResources) FindActiveResource(name string) Disposable {
 func (r *TestResources) setup() error {
 	r.ctx, r.cancel = context.WithCancel(context.Background())
 
-	for dr := r.dequeueResource(); dr != nil; dr = r.dequeueResource() {
-		err := dr.Init(r.ctx)
-		r.pushActiveResource(dr)
+	resourceCount := 0
+	errs := make(chan error)
+	for {
+		dr := r.dequeueResource()
+		if dr == nil {
+			break
+		}
+
+		resourceCount++
+		go func() {
+			err := dr.Init(r.ctx)
+			r.pushActiveResource(dr)
+			errs <- err
+		}()
+	}
+
+	allErrs := make([]error, 0)
+	for i := 0; i < resourceCount; i++ {
+		err := <-errs
 		if err != nil {
-			return err
+			allErrs = append(allErrs, err)
 		}
 	}
-	return nil
+
+	return errors.Join(allErrs...)
```

Comment:
ideally we can use waitgroup here

Reply:
With waitgroup it's harder to collect errors (without causing race conditions for concurrent access to a slice). This does achieve a similar effect while also allowing the collection of errors in a thread-safe way.